packages/peregrine/lib/Apollo/clearWishlistDataFromCache.js

```diff
@@ -0,0 +1,13 @@
+import { deleteCacheEntry } from './deleteCacheEntry';
+
+/**
+ * Deletes references to Customer Wishlists from the apollo cache including *WishlistItem entries
+ *
+ * @param {ApolloClient} client
+ */
+export const clearWishlistDataFromCache = async client => {
```

Comment:
I'm not sure about another utility function here, since I would consider wishlist part of customer data. Fine to keep this, but would be better to have `clearCustomerDataFromCache` call it.

Alternatively, there is a data security pattern we might want to establish. The `clearCustomerDataFromCache` will clear entities that start with `Customer` and I think we've just gotten lucky so far that they followed this pattern. You could instead define `keyFields` for Wishlist and WishlistItem, and this extra utility function wouldn't be needed.

```js
Wishlist: {
    keyFields: ({ id }) => `CustomerWishlist:${id}`
},
WishlistItem: {
    keyFields: ({ id }) => `CustomerWishlistItem:${id}`
},
```

Reply:
Changed according your recommendations.
Thank you!